src/linear_algebra/finsupp.lean

```diff
@@ -313,10 +313,41 @@ finsupp.sum_single_index (f i).map_zero
 
 theorem lsum_symm_apply (f : (α →₀ M) →ₗ[R] N) (x : α) : lsum.symm f x = f.comp (lsingle x) := rfl
 
+section
+variables (M) (R) (X : Type*)
+
+/--
+The type-level equivalence between R-linear maps from `R` to `M`, and points of `M` itself.
+This says that the forgetful functor from `R`-modules to types is representable, by `R`.
+-/
+@[simps]
+def ring_lmap_equiv_self : (R →ₗ[R] M) ≃ M :=
+{ to_fun := λ f, f 1,
+  inv_fun := λ m, { map_smul' := λ r r', by simp [mul_smul], ..(smul_add_hom R M).flip m },
+  left_inv := λ x, by { ext, simp },
+  right_inv := λ x, by { simp } }
```

Comment:
This probably belongs in `linear_algebra/basic`

Reply:
Done!